src/net/sourceforge/kolmafia/textui/Parser.java

```diff
@@ -813,7 +813,7 @@ private Function parseFunction( final Type functionType, final Scope parentScope
 		Scope scope = this.parseBlockOrSingleCommand( functionType, paramList, parentScope, false, false, false );
 
 		result.setScope( scope );
-		if ( !result.assertBarrier() && !functionType.equals( DataTypes.TYPE_VOID ) )
+		if ( !scope.assertBarrier() && !functionType.equals( DataTypes.TYPE_VOID ) )
```

Comment:
What's the rationale for this change? Are we getting rid of UserDefinedFunction.assertBarrier()?

(once I got to the bottom of this review, signs point to "yes". I guess that's okay.)